pypeit/core/wavecal/wv_fitting.py

```diff
@@ -103,33 +103,56 @@ def _bundle(self, **kwargs):
         # Return
         return _d
 
-    def to_hdu(self, hdr=None, add_primary=False, primary_hdr=None, limit_hdus=None):
-        """ Over-ride for force_to_bintbl
+    def to_hdu(self, **kwargs):
+        """
+        Over-ride base-class function to force ``force_to_bintbl`` to always be
+        true.
 
-        See :class:`pypeit.datamodel.DataContainer.to_hdu` for Arguments
+        See base-class :class:`~pypeit.datamodel.DataContainer.to_hdu` for arguments.
+
+        Args:
+            kwargs:
+                Passed directly to the base-class
+                :class:`~pypeit.datamodel.DataContainer.to_hdu`.  If 
+                ``force_to_bintbl`` is present, it is forced to be True.
 
         Returns:
             :obj:`list`, `astropy.io.fits.HDUList`_: A list of HDUs,
             where the type depends on the value of ``add_primary``.
         """
-        return super(WaveFit, self).to_hdu(hdr=hdr, add_primary=add_primary, primary_hdr=primary_hdr,
-                                           limit_hdus=limit_hdus, force_to_bintbl=True)
+        if 'force_to_bintbl' in kwargs:
+            if not kwargs['force_to_bintbl']:
+                msgs.warn(f'{self.__class__.__name__} objects must always use '
+                          'force_to_bintbl = True!')
+            kwargs.pop('force_to_bintbl')
+        return super().to_hdu(force_to_bintbl=True, **kwargs)
 
     @classmethod
-    def from_hdu(cls, hdu, chk_version=True):
+    def from_hdu(cls, hdu, **kwargs):
```

Comment:
Is it possible to simply explicitly put in all the possibilities here. I am in a broad sense opposed to the use of kwargs unless they are absolutely necessary. I guess absolutely necessary could be defined as a situation where the number of parameters is so large that it would be onerous to explicitly name them all, which I doubt to be the case here.

Reply:
In general, I agree that it's best to avoid using `kwargs`.  The issue is that, in cleaning up `DataContainer` and its subclasses, I think we as developers were finding it difficult to know what should be added where.  I changed to using `kwargs` in a few places related to `DataContainer` because all of those keywords are only ever passed directly the base-class method.  That makes it clear what the relevant keywords are (you just have to take the extra step of looking at the base-class function) and it means that any changes made to the base-class method keywords are automatically propagated to the subclasses.  Although I agree this obfuscates the keywords allowed in some of the sub-class methods, I get around this by being very clear in the docstring.